Core/Core/Api/GraphQL/Client.GraphqlCleintOperations/Client.BranchOperations.cs

```diff
@@ -67,10 +74,14 @@ public Task<List<Branch>> StreamGetBranches(string streamId, int branchesLimit =
                         }}                       
                       }}
                     }}",
-      Variables = new { streamId }
-    };
-    var res = await ExecuteGraphQLRequest<StreamData>(request, cancellationToken).ConfigureAwait(false);
-    return res.stream.branches.items;
+        Variables = new { streamId }
+      };
+      var res = await ExecuteGraphQLRequest<StreamData>(request, cancellationToken).ConfigureAwait(false);
+      branches.AddRange(res.stream.branches.items);
+      nextCursor = res.stream.branches.cursor;
+    } while (branches.Count < branchesLimit && nextCursor != null);
```

Comment:
This while loop is questionable, depending on the inputs it could hit our rate-limit quite easily.